library/std/src/net/ip.rs

```diff
@@ -1140,10 +1348,18 @@ impl Ipv6Addr {
     /// assert_eq!(Ipv6Addr::new(0, 0, 0, 0, 0, 0xffff, 0xc00a, 0x2ff).is_unspecified(), false);
     /// assert_eq!(Ipv6Addr::new(0, 0, 0, 0, 0, 0, 0, 0).is_unspecified(), true);
     /// ```
+    ///
+    /// # Stability guarantees
+    ///
+    /// This method's behavior [may by subject to changes][changes] in the
+    /// future, as new IETF RFCs are published.
+    ///
+    /// [changes]: super#stability-guarantees-for-ietf-defined-behavior
     #[rustc_const_stable(feature = "const_ipv6", since = "1.50.0")]
     #[stable(since = "1.7.0", feature = "ip_17")]
     pub const fn is_unspecified(&self) -> bool {
         u128::from_be_bytes(self.octets()) == u128::from_be_bytes(Ipv6Addr::UNSPECIFIED.octets())
+            || if let Some(v4_addr) = self.to_ipv4() { v4_addr.is_unspecified() } else { false }
     }
```

Comment:
Python considers `::` (`Ipv6Addr::UNSPECIFIED`, all bits zero) to be the only unspecified IPv6 address, and `0.0.0.0` to be the only unspecified IPv4 address. I'm not sure what the best approach for Rust here is, as you could indeed argue that `::ffff:0.0.0.0` represents the unspecified `0.0.0.0` address. On the other hand, unspecified == all bits zero makes sense to me, and I don't think `::ffff:0.0.0.0` is defined as unspecified in any protocol.

Reply:
Ok, I'm on board with that reasoning -- I'll remove the v4 check from this method